api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java

```diff
@@ -9,8 +20,47 @@
 @Service
 public class MailServiceImpl implements MailService {
 
+    private final Provider<MandrillApi> mandrillApiProvider;
+    private final Provider<CloudStorageService> cloudStorageServiceProvider;
+    private Provider<WorkbenchConfig> workbenchConfigProvider;
+
+    @Autowired
+    public MailServiceImpl(Provider<MandrillApi> mandrillApiProvider,
+                           Provider<CloudStorageService> cloudStorageServiceProvider,
+                           Provider<WorkbenchConfig> workbenchConfigProvider) {
+        this.mandrillApiProvider = mandrillApiProvider;
+        this.cloudStorageServiceProvider = cloudStorageServiceProvider;
+        this.workbenchConfigProvider = workbenchConfigProvider;
+    }
+
     public void send(Message msg) throws MessagingException {
         Transport.send(msg);
     }
 
+    public void sendWelcomeEmail(String contactEmail, String password, User user) throws MessagingException {
+        String apiKey = cloudStorageServiceProvider.get().readMandrillApiKey();
+        MandrillApiKeyAndMessage keyAndMessage = new MandrillApiKeyAndMessage();
+        keyAndMessage.setKey(apiKey);
+        MandrillMessage msg = buildWelcomeMessage(contactEmail, password, user);
+        keyAndMessage.setMessage(msg);
+        try {
+            mandrillApiProvider.get().send(keyAndMessage);
```

Comment:
I do have a question here. Is it possible for `MandrillMessageStatuses` to indicate some kind of error with a successful call to the `send()` method? If so, we should inspect the results and throw an error if there is one.

Reply:
Clarification. There are actually 2 error cases here. The API can throw an exception, which is handled in this try block. 

The second error case is more subtle. If the response from `mandrillApiProvider.get().send()` indicates that the message was not actually sent, then we need to log that and throw an exception as well. We don't need to ever return the response from `mandrillApiProvider.get().send()`, but we do need to inspect it and deal with it if there was a sending problem.